src/Tribe/Attendee_Registration/Service_Provider.php

```diff
@@ -8,36 +8,128 @@ class Tribe__Tickets__Attendee_Registration__Service_Provider extends tad_DI52_S
+	 */
+	public function add_attendee_registration_shortcode_hook() {
+		/** @var Tribe__Tickets__Attendee_Registration__Shortcode $make */
+		$make = $this->container->make( 'tickets.attendee_registration.shortcode' );
+
+		$make->hook();
+	}
+
+	/**
+	 * Setup Modal Cart Template.
+	 */
+	public function add_attendee_registration_modal_hook() {
+		/** @var Tribe__Tickets__Attendee_Registration__Modal $make */
+		$make = $this->container->make( 'tickets.attendee_registration.modal' );
+
+		$make->hook();
+	}
+
+	/**
+	 * Add attendee-registration rewrite tag.
+	 */
+	public function add_rewrite_tags() {
+		/** @var Tribe__Tickets__Attendee_Registration__Rewrite $make */
+		$make = $this->container->make( 'tickets.attendee_registration.rewrite' );
+
+		$make->add_rewrite_tags();
+	}
+
+	/**
+	 * Sets up the rules required by Event Tickets.
 	 *
-	 * @since 4.9
+	 * @param Tribe__Tickets__Attendee_Registration__Rewrite $rewrite
+	 */
+	public function generate_core_rules( Tribe__Tickets__Attendee_Registration__Rewrite $rewrite ) {
+		/** @var Tribe__Tickets__Attendee_Registration__Rewrite $make */
+		$make = $this->container->make( 'tickets.attendee_registration.rewrite' );
+
+		$make->generate_core_rules( $rewrite );
+	}
+
+	/**
+	 * Generate the Rewrite Rules.
+	 *
+	 * @param WP_Rewrite $wp_rewrite WordPress rewrite that will be modified, pass it by reference (&$wp_rewrite)
 	 */
-	protected function hooks() {
-		add_action( 'plugins_loaded', tribe_callback( 'tickets.attendee_registration.template', 'hook' ) );
-		add_action( 'init', tribe_callback( 'tickets.attendee_registration.shortcode', 'hook' ) );
-		add_action( 'init', tribe_callback( 'tickets.attendee_registration.modal', 'hook' ) );
-		add_action( 'tribe_tickets_pre_rewrite', tribe_callback( 'tickets.attendee_registration.rewrite', 'generate_core_rules' ) );
-		add_action( 'init', tribe_callback( 'tickets.attendee_registration.rewrite', 'add_rewrite_tags' ) );
-		add_filter( 'generate_rewrite_rules', tribe_callback( 'tickets.attendee_registration.rewrite', 'filter_generate' ) );
-		add_filter( 'rewrite_rules_array', tribe_callback( 'tickets.attendee_registration.rewrite', 'remove_percent_placeholders' ), 25 );
-		add_filter( 'tribe_tickets_commerce_paypal_add_to_cart_args', tribe_callback( 'tickets.attendee_registration.meta', 'add_product_delete_to_paypal_url' ), 10, 1 );
+	public function filter_generate( WP_Rewrite $wp_rewrite ) {
+		/** @var Tribe__Tickets__Attendee_Registration__Rewrite $make */
+		$make = $this->container->make( 'tickets.attendee_registration.rewrite' );
+
+		$make->filter_generate( $wp_rewrite );
+	}
+
+	/**
+	 * @param array $rules
+	 *
+	 * @return array
+	 */
+	public function remove_percent_placeholders( array $rules ) {
+		/** @var Tribe__Tickets__Attendee_Registration__Rewrite $make */
+		$make = $this->container->make( 'tickets.attendee_registration.rewrite' );
+
+		return $make->remove_percent_placeholders( $rules );
 	}
 
+	/**
```

Comment:
:no_entry_sign: Doc comment for parameter "$args" missing

<sub>hash: e334297a566a5d07b75635482bbda135</sub>